In [4]:
import math
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import scipy.stats as stats
import statsmodels.formula.api as smf
import statsmodels.api as sm

from math import exp
from scipy.stats import f_oneway
from scipy.stats import chi2_contingency
from scipy.stats import ttest_rel
from scipy.stats import binom
from scipy.stats import poisson
from scipy.stats import ttest_1samp
from scipy.stats import mode
from scipy.stats import ttest_ind
from scipy.stats import norm
from scipy.stats import t

Exercise 1

Consider the following hypothesis test:

H0 : µ = 18

H1 : µ ̸= 18

A sample with n = 48 observations provided a sample mean ¯x = 17 and a sample standard
deviation s = 4.5. It is assumed that the sample is collected from a normal distributed population.

Question I.1 (1)

What is the value of the test statistic (tobs) and what are the critical values with significance
level α = 0.05 (both answers must be correct)?

In [2]:
# Given values
mu = 18  # Null hypothesis mean
x_bar = 17  # Sample mean
s = 4.5  # Sample standard deviation
n = 48  # Sample size
alpha = 0.05  # Significance level

# Degrees of freedom
df = n - 1

# Calculate test statistic (t_obs)
t_obs = (x_bar - mu) / (s / math.sqrt(n))

# Calculate critical t-values for two-tailed test
t_critical_lower = stats.t.ppf(alpha / 2, df)  # Lower critical value
t_critical_upper = stats.t.ppf(1 - alpha / 2, df)  # Upper critical value

# Output results
print("Test Statistic (t_obs):", t_obs)
print("Critical Values (t_critical_lower, t_critical_upper):", (t_critical_lower, t_critical_upper))

Test Statistic (t_obs): -1.539600717839002
Critical Values (t_critical_lower, t_critical_upper): (-2.011740513729766, 2.0117405137297655)


So the awnser must be 1

Question I.2 (2)

From another sample of n = 45 observations, the value of the test statistic (tobs) is -1.74.
Compute the p-value and draw a conclusion using significance level α = 0.05 (both must be
correct)

In [5]:
# Given values
t_obs = -1.74
n = 45
alpha = 0.05

# Degrees of freedom
df = n - 1

# Calculate p-value for two-tailed test
p_value = 2 * t.cdf(t_obs, df)

# Decision
if p_value < alpha:
    conclusion = "Reject the null hypothesis (H0)."
else:
    conclusion = "Fail to reject the null hypothesis (H0)."

# Output
print(f"t-Statistic (t_obs): {t_obs}")
print(f"Degrees of Freedom: {df}")
print(f"p-value: {p_value}")
print(f"Conclusion: {conclusion}")

t-Statistic (t_obs): -1.74
Degrees of Freedom: 44
p-value: 0.08885012244793078
Conclusion: Fail to reject the null hypothesis (H0).


Since the p-value is 0.08885012244793078 we round up and the anwser must be 1. The p-value is 0.0889. We do not reject the null hypothesis

Exerice II

A random sample of n = 30 observations is collected and the sample mean is estimated to be
x¯ = 1.01 and the sample standard deviation s = 0.09.

Question II.1 (3)

What is the 95% confidence interval for the standard deviation?

In [6]:
# Given values
n = 30  # sample size
s = 0.09  # sample standard deviation
alpha = 0.05  # significance level
df = n - 1  # degrees of freedom

# Calculate chi-square critical values
chi2_lower = stats.chi2.ppf(alpha / 2, df)  # lower critical value
chi2_upper = stats.chi2.ppf(1 - alpha / 2, df)  # upper critical value

# Confidence interval for standard deviation
std_lower = np.sqrt((df * s**2) / chi2_upper)  # lower bound
std_upper = np.sqrt((df * s**2) / chi2_lower)  # upper bound
std_lower, std_upper

(0.07167662394863719, 0.12098836613164021)

Again we round up a bit and the awnser is 5

Exercise III

The outcome of an experiment is described by the random variable X, where X has the following
density function:

x     0    1    2   3  4

f(x) 0.17 0.22 0.28 0 0.33

Question III.1 (4)

What is the probability P(X < 3)

In [7]:
# Values for X and corresponding probabilities f(x)
X = [0, 1, 2, 3, 4]
f_x = [0.17, 0.22, 0.28, 0, 0.33]
# Calculate P(X < 3)
P_X_less_3 = sum([f_x[i] for i in range(len(X)) if X[i] < 3])  #change 3 to change the range
print("P(X < 3):", P_X_less_3)

P(X < 3): 0.67
Variance of X: 2.21


The awnser is 5

Question III.2 (5)

The mean of X is 2.10. What is the variance of X?

In [9]:
mean_X = 2.10  # Provided

# Calculate E[X^2]
E_X2 = sum([X[i]**2 * f_x[i] for i in range(len(X))])

# Calculate variance
variance_X = E_X2 - mean_X**2
print("Variance of X:", variance_X)

Variance of X: 2.21


The awnser is 3